test/unit/templateLiterals.spec.ts

```diff
@@ -60,3 +60,29 @@ test.each(["func`noSelfParameter`", "obj.func`noSelfParameter`", "obj[`func`]`no
         `.expectToMatchJsResult();
     }
 );
+
+test.each([
+    ["string", false],
+    ["'string literal type'", false],
+    ["string & unknown", false],
+    ["number", true],
+    ["string | any", true],
+])("template span expect tostring for type (%p) - %p", (type, expectToString) => {
+    util.testFunction`
+        function func(msg: ${type}) {
+            return \`\${msg}\`;
+        }
+
+        // @ts-ignore
+        return func("");
+    `
+        .tap(builder => {
```

Comment:
I would prefer one test case for contains "tostring" and one test case for does not contain "tostring". That will result in a small amount of code duplication, but is a lot easier to read imo.